Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1607,29 +1618,110 @@ - (void)handleInitSuccess {
         // If we find a matching key, configure and show the controller
         if ([keysInParams count]) {
             NSString *key = [[keysInParams allObjects] firstObject];
-            UIViewController <BranchDeepLinkingController> *branchSharingController = self.deepLinkControllers[key];
-            if ([branchSharingController respondsToSelector:@selector(configureControlWithData:)]) {
-                [branchSharingController configureControlWithData:latestReferringParams];
-            }
-            else {
-                [self.preferenceHelper log:FILE_NAME line:LINE_NUM message:
-                    @"[Branch Warning] View controller does not implement configureControlWithData:"];
-            }
-            branchSharingController.deepLinkingCompletionDelegate = self;
-            self.deepLinkPresentingController = [[[UIApplicationClass sharedApplication].delegate window] rootViewController];
             
-            if ([self.deepLinkPresentingController presentedViewController]) {
-                [self.deepLinkPresentingController dismissViewControllerAnimated:NO completion:^{
-                    [self.deepLinkPresentingController presentViewController:branchSharingController animated:YES completion:NULL];
-                }];
+            if([self.deepLinkControllers[key] isKindOfClass:[BranchDeepLinkModel class]]) {
+                BranchDeepLinkModel* deepLinkModel = self.deepLinkControllers[key];
+                UIViewController <BranchDeepLinkingController> *branchSharingController = deepLinkModel.viewController;
```

Comment:
This will crash by using the legacy method that you now list as deprecated. Please be more careful about testing these edge cases